src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/SourceRepositoryExtensions.cs

```diff
@@ -166,17 +169,20 @@ public static Task<SearchResult<IPackageSearchMetadata>> SearchAsync(this Source
                 // filter packages based on allowed versions
                 var updatedPackages = packages.Where(p => allowedVersions.Satisfies(p.Identity.Version));
 
+                cancellationToken.ThrowIfCancellationRequested();
                 var highest = updatedPackages
                     .OrderByDescending(e => e.Identity.Version, VersionComparer.VersionRelease)
                     .FirstOrDefault();
 
+                cancellationToken.ThrowIfCancellationRequested();
                 return highest?.WithVersions(ToVersionInfo(packages, includePrerelease));
```

Comment:
I don't know how expensive you think `highest?.WithVersions(ToVersionInfo(packages, includePrerelease))` might be, but I'd suggest not calling `ThrowIfCancellationRequested` since the method is pretty much done at this point.